lab1/lab1/lab1.cpp

```diff
@@ -0,0 +1,197 @@
+        }
+        previous = current;
+    }
+    file.close();
+    return true;
+
+}
+
+void splitFiles(const std::string& file, const std::string& fileA, const std::string& fileB) {
+    std::ifstream f(file);
+    std::ofstream fA(fileA);
+    std::ofstream fB(fileB);
+
+    if (!f.is_open() || !fA.is_open() || !fB.is_open()) {
+        std::cerr << "Ошибка при открытии файлов." << "\n";
+        return;
+    }
+
+    int number1, number2;
+    bool inFile1 = true;
+
+    if (!(f >> number1)) {
+        std::cerr << "Файл пустой." << "\n";
+        return;
+    }
+
+    while (f >> number2) {
+        if (inFile1) {
+            fA << number1 << " ";
+        }
+        else {
+            fB << number1 << " ";
+        }
+
+        if (number1 <= number2) {
+            number1 = number2;
+        }
+        else {
+            number1 = number2;
+            inFile1 = !inFile1;
+        }
+    }
+
+    if (inFile1) {
+        fA << number1 << " ";
+    }
+    else {
+        fB << number1 << " ";
+    }
+
+    f.close();
+    fA.close();
+    fB.close();
+}
+
+void mergeFiles(const std::string& fileA, const std::string& fileB, const std::string& fileC, const std::string& fileD) {
+    std::ifstream fA(fileA);
+    std::ifstream fB(fileB);
+    std::ofstream fC(fileC);
+    std::ofstream fD(fileD);
+
+    if (!fA.is_open() || !fB.is_open() || !fC.is_open() || !fD.is_open()) {
+        std::cerr << "Ошибка при открытии файлов." << "\n";
+        return;
+    }
+
+    Sequence seq1, seq2;
+    seq1.file = &fA;
+    seq2.file = &fB;
+
+    seq1.readNext();
+    seq2.readNext();
+
+    while (seq1.isValueValid || seq2.isValueValid) {
+        while (!seq1.ended || !seq2.ended) {
+            if (!seq1.ended && (seq2.ended || seq1.value <= seq2.value)) {
+                fC << seq1.value << " ";
+                seq1.readNext();
+            }
+            else if (!seq2.ended) {
+                fC << seq2.value << " ";
+                seq2.readNext();
+            }
+        }
+
+        seq1.ended = !seq1.isValueValid;
+        seq2.ended = !seq2.isValueValid;
+
+        fC.swap(fD);
+    }
+
+    fA.close();
+    fB.close();
+    fC.close();
+    fD.close();
+}
+
+std::string sortFile(const std::string& file,
+    const std::string fileA = "fileA.txt",
+    const std::string fileB = "fileB.txt",
+    const std::string fileC = "fileC.txt",
+    const std::string fileD = "fileD.txt") {
+
+    splitFiles(file, fileA, fileB);
+
+    std::string input1 = fileA;
+    std::string input2 = fileB;
+    std::string output1 = fileC;
+    std::string output2 = fileD;
+
+    while (true) {
+        mergeFiles(input1, input2, output1, output2);
+
+        Sequence seq;
+        std::ifstream currentInput(input2);
+        seq.file = &currentInput;
+        seq.readNext();
+
+        if (!seq.isValueValid) {
+            return output1;
+        }
+
+        std::swap(input1, output1);
+        std::swap(input2, output2);
+    }
```

Comment:
Почему бы не организовать работу цикла следующим образом?

```suggestion
    while (mergeFiles(input1, input2, output1, output2)) {
        std::swap(input1, output1);
        std::swap(input2, output2);
    }
```

Вы ведь в `mergeFiles` всё равно будете открывать файлы и читать оттуда. Нет нужды дублировать эту логику.